src/test/rustdoc/process-termination.rs

```diff
@@ -0,0 +1,37 @@
+// Copyright 2018 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+/// A check of using various process termination strategies
+///
+/// # Examples
+///
+/// ```rust
+/// assert!(true); // this returns `()`, all is well
+/// ```
+///
+/// Papercut: Turbofish to fully specify type
+///
+/// ```rust
+/// Result::Ok::<(), &'static str>(())
```

Comment:
I should say that I wanted to leave this comment on this issue rather than a separate issue or chat because I wanted to make sure that we avoid recommending that users use a syntax that we're not planning on supporting long-term.

Reply:
`Variant::<...>` is *not going anywhere*. `Enum::Variant::<...>` is the mistake we made, we should've moved to `Enum::<...>::Variant`, and ideally base it on `<Enum<...>>::Variant`.
But when there's a single path component, it makes sense to stuck all the parameters on there, IMO.

Reply:
Yeah, I agree, it makes sense... but only in the case of single-component paths. In any case, we're not breaking anything for now; it will all be accepted, though the plan is to lint against `Enum::Variant::<...>` in the future.